federation/pkg/kubefed/init/init.go

```diff
@@ -292,7 +304,7 @@ func (i *initFederation) Run(cmdOut io.Writer, config util.AdminConfig) error {
 
 	fmt.Fprint(cmdOut, "Creating federation control plane service...")
 	glog.V(4).Info("Creating federation control plane service")
-	svc, ips, hostnames, err := createService(cmdOut, hostClientset, i.commonOptions.FederationSystemNamespace, serverName, i.commonOptions.Name, i.options.apiServerAdvertiseAddress, i.options.apiServerServiceType, i.options.dryRun)
+	svc, ips, hostnames, err := createService(cmdOut, hostClientset, i.commonOptions.FederationSystemNamespace, serverName, i.commonOptions.Name, i.options.apiServerAdvertiseAddress, i.options.apiServerNodePortsPort, i.options.apiServerServiceType, i.options.dryRun)
```

Comment:
The concern I had is that you've duplicated the conceptual logic that says "if the node port flag is 0, then do not set it": in the code that validates the argument, and in the code that actually sets it on the object you're creating. If you passed in a pointer, you could default it to nil and, when you're doing the check that the node port is in-range, you could check whether the node port is non-zero and, if so, set the pointer to point to the integer value. Then, the code that sets the node port doesn't need to know about 0 being a special value, but that nil is a special value, which requires less detailed understanding of the semantic meaning of the value being passed in. It makes the checking code a bit more complex, but it localizes all the code that handles checking and understanding the semantics of the flag to one block, and makes the code that sets the nodePort easier to understand without understanding the rest of the code.

The pointer would point to an integer: if you pass in nil, then there would be nothing to set on the NodePort, and if you pass in non-nil, you'd dereference the pointer to get the port value.